### PR TITLE
refactor of int ops (i32/i64) into functor

### PR DESCRIPTION
**Pull request by naturaltransformation**
A simple refactor of the i32 and i64 operators into a single functor

### CODE REVIEW

**Comment by rossberg-chromium** [comment]
lgtm

File: ml-proto/spec/int.ml [hunk 1]
```diff
@@ -0,0 +1,197 @@
+(* WebAssembly-compatible i32 implementation *)
```
Comment (sunfishcode) [inline ml-proto/spec/int.ml]: This comment needs updating.

File: ml-proto/spec/int.ml [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+    op (Rep.add x Rep.min_int) (Rep.add y Rep.min_int)
+
+  (*
+   * Unsigned division and remainder in terms of signed division; algorithm from
+   * Hacker's Delight, Second Edition, by Henry S. Warren, Jr., section 9-3
+   * "Unsigned Short Division from Signed Division".
+   *)
+  let divrem_u n d =
+    if d = Rep.zero then raise Numerics.IntegerDivideByZero else
+      let t = Rep.shift_right d (32 - 1) in
```
Comment (sunfishcode) [inline ml-proto/spec/int.ml]: This 32 - 1 was 64 - 1 in the original i64 code. I don't yet know if this means the algorithm works either way or if we need more tests.

Comment (naturaltransformation) [inline ml-proto/spec/int.ml]: Good catch. git rebase regression unfortunately.

Comment (naturaltransformation) [inline ml-proto/spec/int.ml]: Probably need another test case too. I can propose one in a subsequent PR if needed.

**Comment by sunfishcode** [comment]
That works. Feel free to file a new issue for investigating the div_u/rem_u question and adding more tests if needed, or I can file it too, when I get a chance. Thanks!

**Pull request merged**